alerting/src/main/kotlin/org/opensearch/alerting/TriggersActionThresholdUtils.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.util
+
+object TriggersActionThresholdUtils {
+
+    /**
+     Get threshold value
+     */
+    fun getThreshold(
+        p: TriggersActionThresholdParams,
+        currentTriggerActionSize: Int,
+        triggerMaxActions: Int,
+        triggerTotalMaxActions: Int
+    ): Int {
+        var threshold = currentTriggerActionSize
+        if (triggerTotalMaxActions == 0) {
+            threshold = 0
+        } else if (triggerTotalMaxActions > 0) {
+            threshold = if (triggerMaxActions >= 0) {
+                val temporaryThreshold = if (triggerMaxActions < p.surplusActionCount) triggerMaxActions else p.surplusActionCount
+                if (temporaryThreshold < currentTriggerActionSize) temporaryThreshold else currentTriggerActionSize
+            } else {
+                currentTriggerActionSize
+            }
+        }
```

Comment:
I see room to simplify this logic. Also if we can do an update time check on the dynamic setting value, we can enforce a consistent way to limit the total actions across monitors. Thoughts?

Reply:
> I see room to simplify this logic. Also if we can do an update time check on the dynamic setting value, we can enforce a consistent way to limit the total actions across monitors. Thoughts?

Can i get more context about this update time check on the dynamic setting value?

Reply:
Whenever user updates the max action settings value to a lower number, than the total max actions already configured in the cluster, should the settings update validation check fail and throw an appropriate message to the user? That way we can enforce user to actually reduce the total actions first, before reducing the max limit.